File: public/actions/javascript/upsertData.ts
import { FetchEntries, Findex, FindexKey, IndexedValue, InsertChains, Keyword, Label, Location, UpsertEntries } from "cloudproof_js";

interface IndexedEntry {
  indexedValue: IndexedValue | Location | Keyword;
  keywords: Set<Keyword> | Keyword[] | string[];
}

//
// Index your data
//
export const upsertData = async (
  masterKey: FindexKey,
  label: Label,
  indexedEntries: IndexedEntry[],
  fetchEntries: FetchEntries,
  upsertEntries: UpsertEntries,
  insertChains: InsertChains
): Promise<void> => {
  const { upsert } = await Findex();

  await upsert(masterKey, label, indexedEntries, [], fetchEntries, upsertEntries, insertChains);
};

File: src/actions/java/upsertData.java
import com.cosmian.jna.findex.Findex;
//
// Index your data
//
public static Map<Keyword, Set<Location>> upsertData(byte[] key, byte[] label, Map<IndexedValue, Set<Keyword>> indexedValuesAndWords, Database db) {
  return Findex.upsert(key, label, indexedValuesAndWords, db)
}
